packager/react-packager/src/DependencyResolver/fastfs.js

```diff
@@ -293,16 +293,16 @@ class File {
   }
 
   getFiles() {
-    const files = [];
-    Object.keys(this.children).forEach(key => {
+    const files = Object.keys(this.children).map(key => {
       const file = this.children[key];
       if (file.isDir) {
-        files.push(...file.getFiles());
-      } else {
-        files.push(file);
+        return file.getFiles();
```

Comment:
@cpojer if we call concat many times on huge arrays I'm concerned we'll get bad performance unless V8 has some special optimization for concat